torch/testing/_internal/common_methods_invocations.py

```diff
@@ -9445,14 +9486,42 @@ def ref_pairwise_distance(input1, input2):
            dtypesIfCPU=floating_types_and(torch.int64),
            dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
            sample_inputs_func=sample_inputs_avgpool2d),
+    OpInfo('nn.functional.max_pool1d',
+           aten_name='max_pool1d',
+           supports_autograd=True,
+           supports_out=False,
+           assert_jit_shape_analysis=True,
+           dtypesIfCPU=floating_types(),
+           dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
+           decorators=(
+               # Fails JIT output shape checks
+               DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
+           ),
+           sample_inputs_func=sample_inputs_max_pool),
     OpInfo('nn.functional.max_pool2d',
            aten_name='max_pool2d',
            supports_autograd=True,
+           # Vmap is note happy with non-contiguous (channels_last) inputs
+           check_batched_gradgrad=False,
            supports_out=False,
            assert_jit_shape_analysis=True,
            dtypesIfCPU=floating_types(),
            dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
-           sample_inputs_func=sample_inputs_max_pool2d),
+           sample_inputs_func=sample_inputs_max_pool),
+    OpInfo('nn.functional.max_pool3d',
+           aten_name='max_pool3d',
+           supports_autograd=True,
+           supports_out=False,
+           assert_jit_shape_analysis=True,
```

Comment:
This is interesting, was it added recently? From what I remember, this test was failing the last time I made the PR (https://github.com/pytorch/pytorch/pull/67005/files#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R9461)

Reply:
Do you know how I could fix this shape analysis. I did some light debugging and the shape analysis test was matching super wrong output shapes.

Reply:
Anyway, what is the strategy with OpInfo testing? Do we try to enable them all and then deprecate whatever is failing, or do we disable failing tests with flags? I personally find the former way a bit more future proof. What do you think, @mruberry ?

Reply:
That's a good question, Nik! The thing with shape analysis tests is, that they aren't really implemented yet for these ops. (`max_pool2d` has the implementation though).

Since this is more like an absence of implementation, and not necessarily a failure, hence disabling these tests might be a better option. (IMO)

Please see: https://github.com/pytorch/pytorch/pull/67005#discussion_r733726615 for @eellison's comment.

If you are curious, here is where the shape analysis was added for max_pool2d: https://github.com/pytorch/pytorch/pull/63530

Please let me know if you have any follow up questions.

Reply:
cc @eellison 

If a test is failing and the default is not to test it then I'd stick with that instead of enabling the test and then adding an xFail or skip decorator for it.

More generally if a test is controlled through metadata then I'd prefer using the metadata rather than decorators to skip it.